src/go/rpk/pkg/config/params.go

```diff
@@ -1301,6 +1395,23 @@ func (c *Config) fixSchemePorts() error {
 			return fmt.Errorf("unable to fix admin address %v: unsupported scheme %q", a, scheme)
 		}
 	}
+	for i, a := range p.SR.Addresses {
+		scheme, host, port, err := rpknet.SplitSchemeHostPort(a)
+		if err != nil {
+			return fmt.Errorf("unable to fix schema registry address %v: %w", a, err)
+		}
+		switch scheme {
+		case "":
+			if port == "" {
+				port = strconv.Itoa(DefaultSchemaRegPort)
+			}
+			p.SR.Addresses[i] = net.JoinHostPort(host, port)
+		case "http", "https":
+			continue // keep whatever port exists; empty ports will default to 80 or 443
+		default:
+			return fmt.Errorf("unable to fix schema registry address %v: unsupported scheme %q", a, scheme)
+		}
```

Comment:
I'm surprised this isn't a function somewhere

Reply:
Yeah, I was talking with Travis and we have something ~similar for the Admin API but it follows a different logic since we also use the console URL as a redpanda admin.

There is room for refactoring but it will involve changing more packages and testing the Admin API functions, so I think we can leave it for another PR since this is already too big